### ISSUE TITLE
Make the dedicated `mod_mam` pool size configurable

### ISSUE THREAD

**Issue by lavrin** [esl/MongooseIM#395]
Modules `mod_mam_odbc_async_pool_writer` and `mod_mam_muc_odbc_async_pool_writer` use a hardcoded pool size. Make the size tunable from `ejabberd.cfg`.
See #394 for a discussion.

**Comment by vooolll** [comment]
Should it be a different argument of `mod_mam_odbc_async_pool_writer` or `odbc_pool_size` should be used?

**Comment by lavrin** [comment]
I think it should be an option you pass to the module itself, like:

```erlang
{mod_mam_odbc_async_pool_writer, [{pool_size, 5}, ... other opts]}
```

**Comment by vooolll** [comment]
is it a good idea to get pool options from ejaberd.cfg in `worker_count(_Host) ->` definition ?

**Comment by vooolll** [comment]
Could you review my code for `mod_mam_async_pool_writer` . Is it good enought, it actually solves this problem https://gist.github.com/vooolll/102a03b03f2e72625fa9

**Comment by lavrin** [comment]
Solved with accepting #402.
